internal/encoder/flatFile.go

```diff
@@ -0,0 +1,147 @@
+package encoder
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"github.com/mimiro-io/objectstorage-datalayer/internal/conf"
+	"go.uber.org/zap"
+	"io"
+	"strconv"
+	"time"
+)
+
+// FlatFileDecoder ********************** DECODER ****************************************/
+type FlatFileDecoder struct {
+	backend  conf.StorageBackend
+	logger   *zap.SugaredLogger
+	reader   *io.PipeReader
+	scanner  *bufio.Scanner
+	open     bool
+	closed   bool
+	overhang []byte
+	since    string
+}
+
+func (d *FlatFileDecoder) Read(p []byte) (n int, err error) {
+	buf := make([]byte, 0, len(p))
+	var done bool
+	if len(d.overhang) > 0 {
+		buf = append(buf, d.overhang...)
+		d.overhang = nil
+	}
+
+	if !d.open {
+		d.open = true
+		d.scanner = bufio.NewScanner(d.reader)
+		//start json array and add context as first entity
+		buf = append(buf, []byte("[")...)
+		if n, err, done = d.flush(p, buf); done {
+			return
+		}
+		buf = append(buf, []byte(buildContext(d.backend.DecodeConfig.Namespaces))...)
+		if n, err, done = d.flush(p, buf); done {
+			return
+		}
+	}
+	// append one entity per line, comma separated
+	for d.scanner.Scan() {
+
+		line := d.scanner.Text()
+		//d.logger.Debugf("Got line : '%s'", line)
+		entityProps, err := d.ParseLine(line, d.backend.FlatFileConfig)
+		if err != nil {
+			return 0, err
+		}
+
+		var entityBytes []byte
+		entityBytes, err = toEntityBytes(entityProps, d.backend)
+		if err != nil {
+			return 0, err
+		}
+		buf = append(buf, append([]byte(","), entityBytes...)...)
+		if n, err, done = d.flush(p, buf); done {
+			return 0, err
+		}
+	}
+
+	// Add continuation token
+	entity := map[string]interface{}{
```

Comment:
in a fullsync, the continuatoin token is an empty string. should we add it to the response in that case?